pkg/client/introduce/client.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package introduce
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/google/uuid"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/introduce"
+	"github.com/hyperledger/aries-framework-go/pkg/kms"
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+// provider contains dependencies for the introduce protocol and is typically created by using aries.Context()
+type provider interface {
+	Service(id string) (interface{}, error)
+	KMS() kms.KeyManager
+	InboundTransportEndpoint() string
+	StorageProvider() storage.Provider
+}
+
+// Client enable access to introduce API
+type Client struct {
+	service.Action
+	service.Message
+	service                  service.DIDComm
+	kms                      kms.KeyManager
+	inboundTransportEndpoint string
```

Comment:
Is this required in introduce protocol ?

Reply:
Not sure, anyway I have deleted it!

Reply:
ok, better to avoid adding redundant code. That string is used by one of the DIDExchange protocol message. Hence it was added in DIDExchange client.